server/src/main/java/org/elasticsearch/search/dfs/AggregatedDfs.java

```diff
@@ -29,8 +29,13 @@ public AggregatedDfs(StreamInput in) throws IOException {
         int size = in.readVInt();
         termStatistics = new HashMap<>(size);
         for (int i = 0; i < size; i++) {
-            Term term = new Term(in.readString(), in.readBytesRef());
-            TermStatistics stats = new TermStatistics(in.readBytesRef(), in.readVLong(), DfsSearchResult.subOne(in.readVLong()));
+            // term constructor copies the bytes so we can work with a slice
+            Term term = new Term(in.readString(), in.readSlicedBytesReference().toBytesRef());
+            TermStatistics stats = new TermStatistics(
+                in.readSlicedBytesReference().toBytesRef(),
```

Comment:
Hmm `TermStatistics` doesn't copy these bytes tho, that seems unsafe?

Reply:
🤦 sorry I even checked the `TermStatistics` code last night ... only to misread our code on the second pass ...